src/main/scala/com/webtrends/harness/component/spray/authentication/OAuthDirectives.scala

```diff
@@ -28,7 +28,8 @@ class OAuthHttpAuthenticator[U](val scope: String, val oauthAuthenticator: OAuth
   }
 
   def getChallengeHeaders(httpRequest: HttpRequest) =
-    `WWW-Authenticate`(HttpChallenge(scheme = this.scheme, realm = scope, params = Map.empty)) :: Nil
+    // Returning Basic scheme here to make sure that we return a basic header to get the username/password dialog
+    `WWW-Authenticate`(HttpChallenge(scheme = "Basic", realm = scope, params = Map.empty)) :: Nil
```

Comment:
Permanent, auth goes through both token and basic and we only get a 401 when both fail, however we will only get the right challenge header (the basic one) if we set basic here otherwise we will get the oauth header back instead just because of the order in which the checks happen. So either way we want basic everytime and this makes sure that is the case